### PR TITLE
Add geminilake support in inteltool

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

File: util/inteltool/gpio_names/geminilake.h [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+static const struct gpio_community glk_community_north = {
+	.name		= "----- GPIO Community North -----",
+	.pcr_port_id	= 0xc5,
+	.group_count	= ARRAY_SIZE(glk_community_north_groups),
+	.groups		= glk_community_north_groups,
+};
+
+static const struct gpio_group glk_group_northwest = {
+	.display	= "----- GPIO Group NorthWest -----",
+	.pad_count	= ARRAY_SIZE(glk_group_northwest_names) / 4,
```
Comment (miczyg1) [inline util/inteltool/gpio_names/geminilake.h]:
```suggestion
	.pad_count	= ARRAY_SIZE(glk_group_northwest_names) / 7,
```

glk_group_northwest_names has 7 columns

File: util/inteltool/gpio_names/geminilake.h [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+	.func_count	= 4,
```
Comment (miczyg1) [inline util/inteltool/gpio_names/geminilake.h:273]:
```suggestion
	.func_count	= 7,
```

There are 6 alternative functions in total

File: util/inteltool/gpio_names/geminilake.h [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+};
+
+static const struct gpio_community glk_community_northwest = {
+	.name		= "--- GPIO Community NorthWest ---",
+	.pcr_port_id	= 0xc4,
+	.group_count	= ARRAY_SIZE(glk_community_northwest_groups),
+	.groups		= glk_community_northwest_groups,
+};
+
+static const struct gpio_group glk_group_west = {
```
Comment (miczyg1) [inline util/inteltool/gpio_names/geminilake.h]: there are no west and southwest communities, please change it to audio and SSC communities

File: util/inteltool/gpio_names/geminilake.h (hunk 2 above)
Comment (IgorBagnucki) [reply util/inteltool/gpio_names/geminilake.h:273]: Shouldn't it be 7? NorthWest group can have 7 alternative functions including the default one.

File: util/inteltool/gpio_names/geminilake.h (hunk 1 above)
Comment (IgorBagnucki) [reply util/inteltool/gpio_names/geminilake.h]: I have changed it to 7

File: util/inteltool/gpio_names/geminilake.h (hunk 3 above)
Comment (IgorBagnucki) [reply util/inteltool/gpio_names/geminilake.h]: I have changed it to correct ones

File: util/inteltool/gpio_groups.c [hunk 4]
```diff
@@ -166,6 +167,9 @@ const struct gpio_community *const *get_gpio_communities(struct pci_dev *const s
 		*community_count = ARRAY_SIZE(icelake_pch_h_communities);
 		*pad_stepping = 16;
 		return icelake_pch_h_communities;
+	case PCI_DEVICE_ID_INTEL_CELERON_PENTIUM_SILVER_LPC:
+		*community_count = ARRAY_SIZE(geminilake_communities);
```
Comment (IgorBagnucki) [inline util/inteltool/gpio_groups.c:171]: @miczyg1 Is the default value of `*pad_stepping = 8` correct? I'm not quite sure about that.

File: util/inteltool/gpio_names/geminilake.h (hunk 2 above)
Comment (miczyg1) [reply util/inteltool/gpio_names/geminilake.h:273]: Right, my bad.

File: util/inteltool/gpio_groups.c (hunk 4 above)
Comment (miczyg1) [reply util/inteltool/gpio_groups.c:171]: `*pad_stepping = 16;` is the correct one IIRC. Check the interval between subsequent DW0 registers

File: util/inteltool/inteltool.h [hunk 5]
```diff
@@ -246,6 +246,8 @@ static inline uint32_t inl(unsigned port)
 #define PCI_DEVICE_ID_INTEL_I5000V		0x25d4
 #define PCI_DEVICE_ID_INTEL_I5000P		0x25d8
 
+#define PCI_DEVICE_ID_INTEL_CELERON_PENTIUM_SILVER_LPC 	0x31E8
```
Comment (miczyg1) [inline util/inteltool/inteltool.h]:
```suggestion
#define PCI_DEVICE_ID_INTEL_GLK_LPC 	0x31E8
```

Comment (IgorBagnucki) [reply util/inteltool/inteltool.h]: Changed it.

**Comment by miczyg1** [comment]
@IgorBagnucki we probably need to add `PCI_DEVICE_ID_INTEL_GLK_LPC`
here https://github.com/3mdeb/coreboot/blob/6aa1b300af6d1aad4b2ceb393c84db54b2b1259c/util/inteltool/pcr.c#L110 as well

GPIOs are accessed via P2SB, which we also need to tell where it is to the inteltool

**Comment by miczyg1** [comment]
https://review.coreboot.org/c/coreboot/+/51130

**Pull request closed without merging**